packages/terra-navigation-layout/src/routing/RoutingStackDelegate.js

```diff
@@ -0,0 +1,70 @@
+import PropTypes from 'prop-types';
+
+const supportedAttributes = [
+  'location', 'stackLocation', 'parentPaths', 'goBack', 'goToRoot',
+];
+
+/**
+ * Creates a new RoutingStackDelegate Object instance based on the supportedAttributes.
+ */
+const create = (data) => {
+  const delegate = {};
+  supportedAttributes.forEach((attribute) => {
+    if (data && data[attribute]) {
+      delegate[attribute] = data[attribute];
+    }
+  });
+
+  return Object.freeze(delegate);
+};
+
+/**
+ * Clones an existing RoutingStackDelegate Object instance and overrides any given attributes.
+ */
+const clone = (delegate, data) => {
+  const ancestorDelegate = delegate || {};
+
+  const mergedData = {};
+  supportedAttributes.forEach((attribute) => {
+    mergedData[attribute] = data[attribute] || ancestorDelegate[attribute];
+  });
+
+  return create(mergedData);
+};
+
+/**
+ * Performs a shallow equality on the supportedAttributes to determine the equality of two
+ * RoutingStackDelegate Object instances.
+ */
+const isEqual = (delegateA, delegateB) => {
+  if (!delegateA || !delegateB) {
+    return false;
```

Comment:
If both are null, empty, or undefined, wouldn't they technically be equal?

Reply:
You're right. I placed the direct equality check ahead of this one to cover that use case.